### PR TITLE
Add benchmark for platform views

### LINKED ISSUE #52684: Add benchmark for testing performance of platform views
This benchmark will be similar to the `scroll_perf` benchmark: https://github.com/flutter/flutter/blob/master/dev/benchmarks/complex_layout/README.md

It will consist of a scrollable list of platform views without overlays and with overlays.

cc @cyanglaz, @amirh

### PR DESCRIPTION
**Pull request by blasten**
Fixes https://github.com/flutter/flutter/issues/52684

This PR adds a layout to the complex_layout app, which renders a few dummy platform views and an animated overlay on top of each platform view.  The driver test scrolls the list and gathers information such as the 90th percentile frame build time in millisec.

### CODE REVIEW

File: dev/benchmarks/complex_layout/android/app/src/main/AndroidManifest.xml
Comment (dnfield) [inline dev/benchmarks/complex_layout/android/app/src/main/AndroidManifest.xml]: I think we should avoid touching this file right now - it would probably be better to just create a new benchmark for this rather than touching the existing one, since this is used by several other benchmarks.

I'm particularly worried about what happens if this puts a devicelab device into a bad state right now.

/cc @godofredoc

File: dev/benchmarks/complex_layout/ios/Runner/Info.plist
Comment (dnfield) [inline dev/benchmarks/complex_layout/ios/Runner/Info.plist]: This should be done in a different application.  This will impact other benchmarks besides this one, since it changes the threading model used by the application on iOS.

File: dev/devicelab/manifest.yaml
Comment (dnfield) [inline dev/devicelab/manifest.yaml]: Can this be `linux/android` instead, where we have more capacity?

**Review by dnfield — COMMENTED** [review]
Code overall looks good, I'm worried about impact application changes might have on other benchmarks.

File: dev/benchmarks/complex_layout/ios/Runner/Info.plist
Comment (blasten) [reply dev/benchmarks/complex_layout/ios/Runner/Info.plist]: This will become the default though. cc @cyanglaz what is the status of that?

Comment (cyanglaz) [reply dev/benchmarks/complex_layout/ios/Runner/Info.plist]: Still have several PRs waiting on different people to reviews. It might take some time as people have limited availability working from home. If you want to land this benchmark soon, we should create a different App.

**Comment by blasten** [comment]
I created a separate app in the meanwhile. Adding a new app to the flutter repo involves a non-trivial diff, so I tried to reuse as much of the existing one. Once thread merging becomes the new default, `complex_layout` and `platform_views_layout` can be combined.  PTAL

**Review by cyanglaz — APPROVED** [review]
LGTM! Thanks!
As long as `["linux/android"]` and `["mac/ios"] ` always running on the same android and ios devices.

**Pull request merged**
